### ISSUE TITLE
clang compilation errors

### ISSUE THREAD

**Issue by heeen** [boocs/unreal-clangd#9]
lots of errors in the clangd output tab:
```
I[13:53:57.290] Failed to compile /home/florian/lg/unreal/UnrealEngine/Engine/Source/ThirdParty/libSampleRate/Private/src_sinc.cpp, index may be incomplete
I[13:53:57.302] Failed to compile /home/florian/lg/unreal/UnrealEngine/Engine/Source/Runtime/SlateCore/Private/Rendering/SlateVectorGraphicsCache.cpp, index may be incomplete
I[13:53:57.397] Failed to compile /home/florian/lg/unreal/UnrealEngine/Engine/Plugins/Experimental/PlatformCrypto/Source/PlatformCryptoOpenSSL/Private/PlatformCryptoOpenSSL.cpp, index may be incomplete
I[13:53:57.557] Failed to compile /home/florian/lg/unreal/UnrealEngine/Engine/Plugins/Experimental/PythonScriptPlugin/Source/PythonScriptPlugin/Private/PyConversion.cpp, index may be incomplete
I[13:53:57.608] Failed to compile /home/florian/lg/unreal/UnrealEngine/Engine/Source/Runtime/TimeManagement/Private/TimeManagementModule.cpp, index may be incomplete

```

Ubuntu 23.10, unreal engine built from git, /usr/bin/clangd (16)

**Comment by boocs** [comment]
I tested 5.3.2 full source with the FPS template. I did get the 'Failed to compile' on some project files. I remembered that the FPS template doesn't adhere to IWYU. After adding a few missing includes, to some project cpp files, the messages went away. When I removed the includes the messages reappeared. Full source project files 'seem' to work fine.

It's just that any files in the Unreal directory most likely won't work when opened currently.

I do plan to attempt to fix this.

**Comment by heeen** [comment]
is there a way to see compilation errors from clangd? that might help us in fixing the flags

**Comment by boocs** [comment]
Nope, not if you're looking at a Unreal Source file since this extension doesn't support that but it can support it.(see next post after this).

---
---

You can check to see if it's one of your project's source files that's causing it by opening up every one of your project's source files and seeing if clangd errors occur in your files.

You can do this easily with this VSCode extension that opens every file in a directory(can do this recursively too).

`Be Careful if you have a lot of files. Do this by subdirectory instead.`

Extension is called:
![image](https://github.com/boocs/unreal-clangd/assets/62588629/fbe17f96-d39f-4cb7-916d-ce7010386c66)

Now when I right click on a folder I get these options:
![image](https://github.com/boocs/unreal-clangd/assets/62588629/71961121-8faf-4f6c-bb43-ec9c04b196d5)

This is the FPS Template so I can just choose 'Open All Folder Files'. I did this on the project's Source/ProjectName folder which doesn't have subfolders.

When you open all these files, clangd will start compiling all of them so it may take awhile depending on your computer and how many files you have open.

![image](https://github.com/boocs/unreal-clangd/assets/62588629/d94cb88c-f5f1-4bff-a3e0-442dbb914492)

Even though this project compiles correctly, clangd sees an error in one of the files. This is because clangd adheres strictly to IWYU which Epic says is the correct way for Unreal Projects. 

This file is missing an include. Once it's added the error goes away.

**Comment by boocs** [comment]
This extension does now support Unreal Source files
